packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,212 @@
+
+/**
+ * Properties of product version (also known as a provisioning artifact).
+ */
+export interface ProductVersion {
+  /**
+   * The description of the product version
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Whether the specified product template will be validated by CloudFormation. If turned off, an invalid template configuration can be stored.
+   * @default true
+   */
+  readonly validateTemplate?: boolean;
+
+  /**
+   * The S3 template that points to the provisioning version template
+   */
+  readonly cloudFormationTemplate: CloudFormationTemplate;
+
+  /**
+   * The name of the product version.
+   * @default - No product version name provided
+   */
+  readonly name?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface CloudFormationProductProps {
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The configuration of the product version.
+   */
+  readonly productVersions: ProductVersion[];
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the product.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * The distributor of the product.
+   * @default - No distributor provided
+   */
+  readonly distributor?: string;
+
+  /**
+   * Whether to give provisioning artifacts a new unique identifier when the product attributes or provisioning artifacts is updated
+   * @default false
+   */
+  readonly replaceProvisioningArtifacts?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+}
+
+/**
+ * Abstract class for Service Catalog Product.
+ */
+export abstract class Product extends ProductBase {
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param productArn Product Arn
+   */
+  public static fromProductArn(scope: Construct, id: string, productArn: string): IProduct {
+    const arn = Stack.of(scope).parseArn(productArn);
+    const productId = arn.resourceName;
+
+    if (!productId) {
+      throw new Error('Missing required Portfolio ID from Portfolio ARN: ' + productArn);
+    }
+
+    class Import extends ProductBase {
+      public readonly productId = productId!;
+      public readonly productArn = productArn;
+    }
+    return new Import(scope, id);
```

Comment:
You can save one line by using an anonymous class:

```suggestion
    return new class extends ProductBase {
      public readonly productId = productId!;
      public readonly productArn = productArn;
    }(scope, id);
```